pkg/server/sandbox_remove.go

```diff
@@ -17,15 +17,76 @@ limitations under the License.
 package server
 
 import (
-	"errors"
+	"fmt"
+	"os"
 
+	"github.com/docker/docker/pkg/truncindex"
+	"github.com/golang/glog"
 	"golang.org/x/net/context"
 
+	"github.com/containerd/containerd/api/services/execution"
+
 	"k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
 )
 
 // RemovePodSandbox removes the sandbox. If there are running containers in the
 // sandbox, they should be forcibly removed.
-func (c *criContainerdService) RemovePodSandbox(ctx context.Context, r *runtime.RemovePodSandboxRequest) (*runtime.RemovePodSandboxResponse, error) {
-	return nil, errors.New("not implemented")
+func (c *criContainerdService) RemovePodSandbox(ctx context.Context, r *runtime.RemovePodSandboxRequest) (retRes *runtime.RemovePodSandboxResponse, retErr error) {
+	glog.V(2).Infof("RemovePodSandbox for sandbox %q", r.GetPodSandboxId())
+	defer func() {
+		if retErr == nil {
+			glog.V(2).Info("RemovePodSandbox returns successfully")
+		}
+	}()
+
+	// Convert potential truncated id to full id.
+	id, err := c.sandboxIDIndex.Get(r.GetPodSandboxId())
+	if err != nil {
+		if err == truncindex.ErrNotExist {
+			// Do not return error if the id doesn't exist.
+			return &runtime.RemovePodSandboxResponse{}, nil
+		}
+		return nil, fmt.Errorf("failed to find sandbox id %q: %v", r.GetPodSandboxId(), err)
+	}
+
+	// TODO(random-liu): [P2] Remove all containers in the sandbox.
+
+	// Return error if sandbox container is not fully stopped.
+	_, err = c.containerService.Info(ctx, &execution.InfoRequest{ID: id})
+	if err != nil && !isContainerdContainerNotExistError(err) {
+		return nil, fmt.Errorf("failed to get sandbox container info for %q: %v", id, err)
+	}
```

Comment:
We should continue cleaning up the following stuff. If last remove failed, we could finish cleanup this time.